### PR TITLE
Setting player modelling 

### LINKED ISSUE #1: Setting player modelling
## Description

* :star2: Goal: Being able to choose player settings without changing the code.
* :arrow_right: Making the difference at the dataset level (when loading and encoding the game)

## Acceptance Criterion

* [ ] The code is documented
* [x] Utilities and class test are written
* [x] The code was reviewed

## Tasks

* [x] Make a second dataset class
* [ ] Test game sequence in case of a loss or win

### PR DESCRIPTION
**Pull request by Xmaster6y**
Closes #1

### CODE REVIEW

File: src/utils/translate.py
Comment (sej2020) [inline src/utils/translate.py:1]: The Stockfish board eval metric has to have access to the chess.Board() boards, not the board_tensors (by necessity, not design choice). I think the best way to integrate that is to return a list of the chess boards at every time step along with move_indices, end_rewards, and board_tensors. Could you implement this in the encode_seq() function and then modify the downstream consumers of encode_seq to handle the additional output? Let me know if you have a better way to implement this

Comment (sej2020) [inline src/utils/translate.py:1]: Line 12: The Stockfish board eval metric has to have access to the chess.Board() boards, not the board_tensors (by necessity, not design choice). I think the best way to integrate that is to return a list of the chess boards at every time step along with move_indices, end_rewards, and board_tensors. Could you implement this in the encode_seq() function and then modify the downstream consumers of encode_seq to handle the additional output? Let me know if you have a better way to implement this

File: src/utils/translate.py [hunk 1]
```diff
@@ -92,28 +93,34 @@ def format_inputs(
     timesteps = torch.arange(start=window_start, end=window_start + window_size, device=device).reshape(1, window_size)
     attention_mask = torch.ones(1, window_size, device=device, dtype=torch.float32)  # Needed for padding
```
Comment (sej2020) [inline src/utils/translate.py:94]: Are you sure that this attention mask accomplishes anything? After looking through the modeling_decision_transformer.py code, it seems to me that 1 is subtracted from the attention mask, making an array of zeros, which are then added to the attention weights. This effectively changes nothing. I'm not sure of the original intention of this attention mask, but I think we also have a problem where there is no autoregressive mask applied to the attention weights. The attention heads can then attend to states further on in the sequence.

**Review by sej2020 — CHANGES_REQUESTED** [review]

File: src/utils/translate.py (hunk 1 above)
Comment (Xmaster6y) [reply src/utils/translate.py:93]: It is the default setting, so not compulsory (I can remove it), but it could be useful for other experiments to not attend the whole sequence.

Comment (Xmaster6y) [reply src/utils/translate.py:93]: For the rest, I don't see how you jumped to this conclusion. We can discuss this on Discord.

File: src/utils/translate.py
Comment (Xmaster6y) [reply src/utils/translate.py:1]: Overkilled, since with the last board state you also get the move stack so you can basically explore the whole history of the board.

**Review by sej2020 — APPROVED** [review]
